wzdx/raw_to_standard/icone.py

```diff
@@ -0,0 +1,223 @@
+            "header": {
+                "description": pd.get("incident/description", default=""),
+                "start_timestamp": pd.get("incident/starttime", date_tools.get_unix_from_iso_string, default=None),
+                "end_timestamp": pd.get("incident/endtime", date_tools.get_unix_from_iso_string, default=None)
+            },
+            "detail": {
+                "road_name": pd.get("incident/location/street"),
+                "road_number": pd.get("incident/location/street"),
+                "direction": get_direction(pd.get("incident/location/street"), pd.get("incident/location/polyline"))
+            },
+            "additional_info": {
+                "devices": devices,
+                "directionality": pd.get("incident/location/direction")
+            }
+        }
+    }
+
+
+def get_direction(street, coords):
+    direction = wzdx_translator.parse_direction_from_street_name(street)
+    if not direction:
+        direction = polygon_tools.get_road_direction_from_coordinates(coords)
+    return direction
+
+
+# function to parse polyline to geometry line string
+def parse_icone_polyline(polylinestring):
+    if not polylinestring or type(polylinestring) != str:
+        return None
+    # polyline rightnow is a list which has an empty string in it.
+    polyline = polylinestring.split(',')
+    coordinates = []
+    for i in range(0, len(polyline)-1, 2):
+        try:
+            coordinates.append([float(polyline[i + 1]), float(polyline[i])])
+        except ValueError as e:
+            logging.warning('failed to parse polyline!')
+            return []
+    return coordinates
+
+
+# function to validate the incident
+def validate_incident(incident):
+    if not incident or (type(incident) != dict and type(incident) != OrderedDict):
+        logging.warning('incident is empty or has invalid type')
+        return False
+
+    id = incident.get("@id")
+
+    location = incident.get('location')
+    if not location:
+        logging.warning(
+            f'Invalid incident with id = {id}. Location object not present')
+        return False
+
+    polyline = location.get('polyline')
+    coords = parse_icone_polyline(polyline)
+    street = location.get('street', '')
+
+    starttime_string = incident.get('starttime')
+    endtime_string = incident.get('endtime')
+    description = incident.get('description')
+    creationtime = incident.get('creationtime')
+    updatetime = incident.get('updatetime')
+    direction = get_direction(street, coords)
+    if not direction:
+        logging.warning(
+            f'Invalid incident with id = {id}. unable to parse direction from street name or polyline')
+        return False
+    required_fields = [location, polyline, coords, street,
+                       starttime_string, description, creationtime, updatetime, direction]
+    for field in required_fields:
+        if not field:
+            logging.warning(
```

Comment:
We can probably pull direction from this list since we check it above and don't include it in the explainer required text below.

Reply:
good call, done